dev-packages/e2e-tests/run.ts

```diff
@@ -68,33 +78,38 @@ async function run(): Promise<void> {
       registrySetup();
     }
 
-    await asyncExec('pnpm clean:test-applications', { env, cwd: __dirname });
     await asyncExec('pnpm cache delete "@sentry/*"', { env, cwd: __dirname });
 
     const testAppPaths = appName ? [appName.trim()] : globSync('*', { cwd: `${__dirname}/test-applications/` });
 
-    console.log(`Runnings tests for: ${testAppPaths.join(', ')}`);
+    console.log(`Running tests for: ${testAppPaths.join(', ')}`);
     console.log('');
 
     for (const testAppPath of testAppPaths) {
       const originalPath = resolve('test-applications', testAppPath);
       const tmpDirPath = await mkdtemp(join(tmpdir(), `sentry-e2e-tests-${appName}-`));
+      tempDirs.add(tmpDirPath);
 
-      await copyToTemp(originalPath, tmpDirPath);
-      const cwd = tmpDirPath;
-
-      console.log(`Building ${testAppPath} in ${tmpDirPath}...`);
-      await asyncExec('pnpm test:build', { env, cwd });
+      try {
+        await copyToTemp(originalPath, tmpDirPath);
+        const cwd = tmpDirPath;
 
-      console.log(`Testing ${testAppPath}...`);
-      await asyncExec('pnpm test:assert', { env, cwd });
+        console.log(`Building ${testAppPath} in ${tmpDirPath}...`);
+        await asyncExec('pnpm test:build', { env, cwd });
 
-      // clean up (although this is tmp, still nice to do)
-      await rm(tmpDirPath, { recursive: true });
+        console.log(`Testing ${testAppPath}...`);
+        await asyncExec('pnpm test:assert', { env, cwd });
+      } finally {
+        // clean up (although this is tmp, still nice to do)
+        await rm(tmpDirPath, { recursive: true });
+        tempDirs.delete(tmpDirPath);
+      }
     }
   } catch (error) {
     console.error(error);
     process.exit(1);
+  } finally {
+    await cleanup();
```

Comment:
I do not think we want to cleanup when tests fail - the reason being that it makes debugging stuff super hard! Right now, if a test fails you can cd into the tmp dir and debug the actual application state properly. Since all the code is in a tmp dir, it should (in theory) be cleaned up by the OS anyhow after some time 🤔

Reply:
alternative idea: What if we prefix the temp dirs we create somehow (e.g. sentry-test-…), and then make sure that running `yarn clean` also cleans the tmp dirs?

Reply:
> alternative idea: What if we prefix the temp dirs we create somehow (e.g. sentry-test-…), and then make sure that running `yarn clean` also cleans the tmp dirs? 

Yes, why not.